andela/ah-orcas
# #161254662 Create user profile upon registration

#### What does this PR do?
```
- creates a profile once a user registers
- a user can retrieve his/her profile
- a user can retrieve all users profiles
- implement update user profile

```
#### Description of Task to be completed?
```
User profile should be automatically created once the user registers
Users who are authenticated should be able to view other profiles but not edit them
An authenticated user should be able to view another user's profile but not edit it
 ```
#### How should this be manually tested?

- create a python virtual environment in your local machine and start it.
```
$ virtualenv env
$ source env/bin/activate
$ cd env
```
- Clone this repo.
``` 
$ git clone https://github.com/andela/ah-orcas.git
```
- Switch to this branch
``` 
$  git checkout  ft-user-profile-16125466
```
- Create a .env file and run it. Check sample.env file to configure yours.
```
$ source .env
```
- Run the app
```
$ python manage.py runserver
```
- Test the endpoint as per the README.md

**Any background context you want to provide?**

```All endpoints require authentication i.e jwt token```

**What are the relevant pivotal tracker stories?**
[#161254662](https://www.pivotaltracker.com/story/show/161254662)
